### ISSUE TITLE
No connection re-use on HTTP executor

### ISSUE THREAD

**Issue by spy16** [distribworks/dkron#904]
**Is your feature request related to a problem? Please describe.**
The HTTP executor seems to be creating new `http.Client` [for every job execution](https://github.com/distribworks/dkron/blob/master/builtin/bins/dkron-executor-http/http.go#L97). This causes 2 types of problems:
1. (Minor) Poor response times or job execution times because new connections are created every time instead of utilising connection pool maintained by `http.Client`
2. (Major) If the server supports Keep Alive connections, repeated job executions end up creating multiple persistent idle connections to the target host eventually causing file-descriptor limit breach on the server side. (If the server handles the timeout on idle connections properly or sends a `Connection: close` on every response, this is handled. But it would still be ideal to handle this properly from the client)

**Describe the solution you'd like**

Re-use the `http.Client` instance in the HTTP executor and allow configuring `MaxIdleConns` and `MaxIdleConnsPerHost ` configs of the http client. (Handling TLS configs would be slightly tricky I think -- But may be it would at least help if the client instances can be cached against Job ID)

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by ncsibra** [comment]
I agree with this issue, we had a goroutine/memory leak, because the TCP connections were kept alive, but the client was not reused, so the resources weren't freed on the server side.  
A `Connection: close` header solves this, but it shouldn't be necessary if dkron reuses the client.

**Comment by vcastellm** [comment]
I will leave this open to tackle the connection re-use at some point.

**Comment by vcastellm** [comment]
Fixed
